### PR TITLE
Support for settings fly-in animation

### PR DESCRIPTION
**Pull request by collinjackson**

### CODE REVIEW

File: sky/sdk/lib/widgets/navigator.dart [hunk 1]
```diff
@@ -35,6 +40,92 @@ class RouteState extends RouteBase {
...
+  Function onDismissed;
+
+  AnimatedType<Point> _position;
+  AnimatedType<double> _opacity;
+  AnimationPerformance _performance;
+
+  void initState() {
+    _position = new AnimatedType<Point>(_kTransitionStartPoint)
+      ..end = Point.origin
+      ..curve = easeOut;
```
Comment (abarth) [inline sky/sdk/lib/widgets/navigator.dart]: You can just pass these as constructor arguments.

File: sky/sdk/lib/widgets/navigator.dart [hunk 2]
```diff
@@ -35,6 +40,92 @@ class RouteState extends RouteBase {
...
+    _opacity = new AnimatedType<double>(0.0, end: 1.0)
+      ..curve = easeOut;
```
Comment (abarth) [inline sky/sdk/lib/widgets/navigator.dart]: ditto

File: sky/sdk/lib/widgets/navigator.dart [hunk 3]
```diff
@@ -35,6 +40,92 @@ class RouteState extends RouteBase {
...
+  }
+
+  void syncFields(Transition source) {
+    content = source.content;
+    if (direction != source.direction) {
+      direction = source.direction;
+      _start();
+    }
+    onDismissed = source.onDismissed;
+    super.syncFields(source);
```
Comment (abarth) [inline sky/sdk/lib/widgets/navigator.dart]: interactive ?

File: sky/sdk/lib/widgets/navigator.dart [hunk 4]
```diff
@@ -115,6 +204,26 @@ class Navigator extends StatefulComponent {
...
+      if (i == 0) {
+        visibleRoutes.add(content);
+        continue;
+      }
+      if (content == null)
+        continue;
+      Transition transition = new Transition(content: content)
+        ..direction = (i <= state.historyIndex) ? TransitionDirection.forward : TransitionDirection.reverse
+        ..interactive = (i == state.historyIndex)
+        ..onDismissed = () {
```
Comment (abarth) [inline sky/sdk/lib/widgets/navigator.dart]: The usual pattern is to pass these as constructor arguments.

**Comment by abarth** [comment]
LGTM

File: sky/sdk/example/widgets/navigation.dart [hunk 5]
```diff
@@ -21,8 +21,10 @@ List<Route> routes = [
         new RaisedButton(
           child: new Text('START ADVENTURE'),
           onPressed: () => navigator.pushNamed('adventure')
-        )
-      ])
+        )],
```
Comment (Hixie) [inline sky/sdk/example/widgets/navigation.dart]: This ends up not aligned. I think you really want the children indented 4 from the 'child', then the other arguments (and the closing "]") indented 2.

**Pull request merged**